command/src/main/java/com/jagrosh/jdautilities/command/ContextMenu.java

```diff
@@ -150,7 +152,17 @@ public CommandData buildCommandData()
         else
             data.setDefaultPermissions(DefaultMemberPermissions.enabledFor(this.userPermissions));
 
-        data.setGuildOnly(this.guildOnly);
+        Set<InteractionContextType> contexts = getContexts();
+
+        // manually set to true
+        if (this.guildOnly == null) {
+            // do nothing!!! nothing!!!!
+        } else if (this.guildOnly) {
+            // remove bot dm from contexts
+            contexts.remove(InteractionContextType.BOT_DM);
+        } else if (!this.guildOnly) {
```

Comment:
Pretty sure a simple `} else {` should work here given you already check for null and if it is true. And since it's a boolean can there only be false be left by the logic.